extractors-opendronemap/terraref-opendronemap.py

```diff
@@ -107,14 +205,58 @@ def dosetup(self, odm_args):
         TerrarefExtractor.setup(self, sensor=self.sensor_name)
         OpenDroneMapStitch.dosetup(self, odm_args)
 
+    def find_timestamp(self, resource, text):
+        """Looks up a timestamp based upon EXIF data. Uses default mechanisms if a
+           timestamp can't be found.
+
+        Args:
+            resource(dict): dictionary containing the resources associated with the request
+            text(str): optional text string to search for a time stamp if we can't use EXIF data
+
+        Return:
+            A timestamp in ISO 8601 long format or None if one isn't found
+        """
+        # pylint: disable=too-many-nested-blocks
+        first_stamp = None
+        try:
+            # Get all the image files
+            paths = list()
+            for localfile in resource['local_paths']:
+                # deal with mounted/local files
+                if localfile.lower().endswith('.jpg'):
+                    paths.append(localfile)
+                else:
+                    # deal with downloaded files
+                    for image in resource['files']:
+                        if 'filepath' in image and image['filepath'] == localfile:
+                            if image['filename'].lower().endswith('.jpg'):
+                                paths.append(image['filename'])
+
+            # Find a timestamp to use by looking at EXIF data
+            for input_path in paths:
+                tags_dict = piexif.load(input_path)
+                if tags_dict and "Exif" in tags_dict:
+                    cur_stamp = exif_tags_to_timestamp(tags_dict["Exif"])
+                    if cur_stamp:
+                        first_stamp = cur_stamp if first_stamp is None or cur_stamp < first_stamp \
+                                                                                else first_stamp
+
+        except Exception as ex:     # pylint: disable=broad-except
```

Comment:
Ditto re narrowing.

Reply:
The method already returns None so the caller should be able to handle that value. I'm leaving it up to the caller to determine what to do if a datetime is not available (for any reason).

Reply:
There are many places to find a timestamp, this is just one. If it doesn't work out, it'll get located at somewhere else. It's a tradeoff between coming to a halt due to one timestamp not being found, and continuing along with a possibly different timestamp. The bigger issue is that there should be one source for the timestamp, not several.